src/hooks/checkout.ts

```diff
@@ -620,36 +621,45 @@ export const useCheck = (
     JSON.stringify(shipping),
   ])
 
-  const placeOrder = useCallback(
-    async (paymentType: 'perpetual' | 'subscription' | 'groupBuying', invoice: InvoiceProps) => {
-      setOrderPlacing(true)
-      return axios
-        .post<{ code: string; message: string; result: { id: string } }>(
-          `${process.env.REACT_APP_API_BASE_ROOT}/tasks/order`,
-          {
-            paymentModel: { type: paymentType },
-            productIds,
-            discountId,
-            memberId,
-            shipping,
-            invoice,
-            options,
-          },
-          {
-            headers: { authorization: `Bearer ${authToken}` },
-          },
-        )
-        .then(({ data: { code, result, message } }) => {
-          if (code === 'SUCCESS') {
-            return result.id
-          } else {
-            throw new Error(message)
-          }
-        })
-        .finally(() => setOrderPlacing(false))
-    },
-    [authToken, discountId, memberId, options, productIds, shipping],
-  )
+  const placeOrder = async (
+    paymentType: 'perpetual' | 'subscription' | 'groupBuying',
+    invoice: InvoiceProps,
+  ) => {
+    setOrderPlacing(true)
+    const {
+      data: { code, message, result },
+    } = await Axios.post<{
+      code: string
+      message: string
+      result: {
+        orderId: string
+        totalAmount: number
+        paymentNo: string | null
+        payToken: string | null
+        products: { name: string; price: number }[]
+        discounts: { name: string; price: number }[]
+      }
+    }>(
+      `${process.env.REACT_APP_API_BASE_ROOT}/order/create`,
+      {
+        memberId,
+        paymentModel: { type: paymentType },
+        productIds,
+        discountId,
+        shipping,
+        invoice,
+        options,
+      },
+      {
+        headers: { authorization: `Bearer ${authToken}` },
+      },
+    )
+    if (code === 'SUCCESS') {
+      return result
+    } else {
+      throw new Error('create order failed: ' + message)
```

Comment:
I think you need to set orderPlacing to false here because there's no place to do. 
And it will cause the Button in AppointmentPlanAppointmentModal:337 to keep loading and can't be clicked.

Reply:
modified